lib/prometheus_exporter/server/web_collector.rb

```diff
@@ -62,11 +62,16 @@ def observe(obj)
       }
       custom_labels = obj['custom_labels']
       labels = custom_labels.nil? ? default_labels : default_labels.merge(custom_labels)
+      http_labels = labels.merge(status: obj["status"])
 
-      @http_requests_total.observe(1, labels.merge(status: obj["status"]))
+      @http_requests_total.observe(1, http_labels)
 
       if timings = obj["timings"]
-        @http_duration_seconds.observe(timings["total_duration"], labels)
+        if obj['http_labels_on_duration']
```

Comment:
This kind of feels like the responsibility of the collector, no need to send this extra information downstream. Perhaps we allow specifying config for collector initialization?